src/main/java/org/spongepowered/api/data/key/KeyFactory.java

```diff
@@ -70,18 +72,43 @@ private KeyFactory() {}
      * @param id The id for the new key
      * @param name The name for the new key
      * @return The generated key
+     * @deprecated use {@link #makeSingleKey(TypeToken, TypeToken, DataQuery, CatalogKey, String)}
      */
+    @Deprecated
```

Comment:
Same here, you already create the key for them...I see no reason to just outright remove the method.

Reply:
I personally don't like that `CatalogKey` has the registration id but we still have the name separate. Perhaps we could merge the two in this case? Still feels cluttered. We may want to move the KeyFActory to use builders instead at this point with the amount of arguments.